python/cudf/cudf/core/dataframe.py

```diff
@@ -1642,7 +1651,10 @@ def drop(self, labels=None, axis=None, columns=None, errors="raise"):
             if isinstance(target, (str, numbers.Number))
             else list(target)
         )
-        outdf = self.copy()
+        if inplace:
+            outdf = self
+        else:
+            outdf = self.copy()
```

Comment:
Does Pandas return this by deep copy? We could get away with a shallow copy here but we should keep the semantics if possible.

Reply:
Confirmed it does return by deep copy.